### ISSUE TITLE
Rails/MatchRoute error

### ISSUE THREAD

**Issue by luigi** [rubocop-hq/rubocop#8380]
## Expected behavior

No error.

## Actual behavior

Output:

```
1 error occurred:
An error occurred while Rails/MatchRoute cop was inspecting /.../pandemic-ebt-ca/config/routes.rb:18:14.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop-hq/rubocop/issues

Mention the following information in the issue report:
0.88.0 (using Parser 2.7.1.4, rubocop-ast 0.2.0, running on ruby 2.6.6 x86_64-darwin19)
RuboCop failed!
```

I reverted back to Rubocop 0.87.1, but the problem persisted.

## Steps to reproduce the problem

Run rubocop on the following routes.rb file:

https://github.com/codeforamerica/pandemic-ebt-ca/blob/master/config/routes.rb#L17-L19

```ruby
17            { get: :edit, put: :update }.each do |method, action|
18              match "/#{controller_class.to_param}",
19                    action: action,
```

## RuboCop version

```
0.88.0 (using Parser 2.7.1.4, rubocop-ast 0.2.0, running on ruby 2.6.6 x86_64-darwin19)
```
